core/tools/files_tool.py

```diff
@@ -0,0 +1,414 @@
+        Show the contents of the file at path relative to current dir
+        """
+        if not os.path.exists(self.base_path):
+            return self.fail_response("Base path does not exist")
+        effective_path = self._get_effective_path(path)
+        if not os.path.commonpath([self.base_path, effective_path]) == os.path.normpath(
+            self.base_path
+        ):
+            return ToolResult(
+                success=False, output="Path is not within the base path", exit_code=1
+            )
+        if os.path.exists(effective_path):
+            with open(effective_path, "r") as file:
+                text = file.read()
+            return self.success_response(text)
+        else:
+            return self.fail_response("File does not exist")
+
+    def read_directory_contents(self, path: str, depth: int = 3) -> ToolResult:
+        """
+        List all files and directories at the given path, including their contents, while excluding certain files and directories as specified in file_utils.py, up to a specified depth.
+        """
+        if not os.path.exists(self.base_path):
+            return self.fail_response("Base path does not exist")
+        effective_path = self._get_effective_path(path)
+        if not os.path.exists(effective_path) or not os.path.isdir(effective_path):
+            return self.fail_response("Directory does not exist or is not a directory")
+
+        try:
+            directory_contents = {}
+            for root, dirs, files in os.walk(effective_path):
+                # Calculate current depth
+                current_depth = os.path.relpath(root, effective_path).count(os.sep)
+                if current_depth > depth:
+                    dirs[:] = []  # Don't go any deeper
+                else:
+                    # Apply exclusions for directories
+                    dirs[:] = [d for d in dirs if d not in EXCLUDED_DIRS and not _should_exclude(self.base_path, os.path.join(root, d))]
+                    # Apply exclusions for files
+                    files = [f for f in files if f not in EXCLUDED_FILES and not any(f.endswith(ext) for ext in EXCLUDED_EXT) and not _should_exclude(self.base_path, os.path.join(root, f))]
+                    # Append relative paths of files and directories to contents and read file contents
+                    for file in files:
+                        file_path = os.path.join(root, file)
+                        relative_file_path = os.path.relpath(file_path, effective_path)
+                        if not _should_exclude(self.base_path, file_path):
+                            try:
+                                with open(file_path, 'r', encoding='utf-8') as file_content:
+                                    directory_contents[relative_file_path] = file_content.read()
+                            except UnicodeDecodeError:
+                                # Skip files that cannot be decoded with UTF-8
+                                continue
```

Comment:
The continue is at the end of the loop right? So it's redundant, can be a `pass`